### ISSUE TITLE
Include assignments in consistent-this

### ISSUE THREAD

**Issue by zmthy** [eslint/eslint#1513]
I tend to group multiple variable declarations together without assigning to them, and perform the assignments underneath. The consistent-this rule currently only checks variable declarations, and forces the aliasing variable to be assigned in its declaration. This means that the following is rejected:

``` js
var foo, self;
self = this;
```

While this is accepted:

``` js
var foo, that;
that = this;
```

It would be nice if the rule considered all assignments to identifiers, and had an option to allow empty variable declarations. I've implemented this change [here](https://github.com/zmthy/eslint/commit/0fcd84d811a0525b7783b265f99492110133cd24), and will submit a pr if there's interest.

**Comment by nzakas** [comment]
Sorry, lost track of this. Definitely see your point, we should fix this.
